rackcity/api/serializers/asset_serializers.py

```diff
@@ -171,6 +156,49 @@ def power_port_connection(self, asset, port_number):
         return pdu_port.left_right+str(pdu_port.port_number)
 
 
+class RecursiveAssetCPSerializer(serializers.ModelSerializer):
+    """
+    Recursively serializes all fields on AssetCP model, where model and
+    rack fields are defined recursively (by all of their respective fields).
+    """
+    model = ITModelSerializer()
+    rack = RackSerializer()
+    mac_addresses = serializers.SerializerMethodField()
+    power_connections = serializers.SerializerMethodField()
+    network_connections = serializers.SerializerMethodField()
+    network_graph = serializers.SerializerMethodField()
+
+    class Meta:
+        model = AssetCP
+        fields = (
+            'id',
+            'asset_number',
+            'hostname',
+            'model',
+            'rack',
+            'rack_position',
+            'owner',
+            'comment',
+            'mac_addresses',
+            'network_connections',
+            'network_graph',
+            'power_connections',
```

Comment:
could be useful to add `change_plan` as a field to the serializer and display this on the front end - but can be added later if we decide we need it

Reply:
t'has been added